public/app/percona/add-instance/components/AddRemoteInstance/FormParts/NodesAgents/NodesAgents.tsx

```diff
@@ -0,0 +1,89 @@
+import { FormApi } from "final-form";
+import React, { FC, useCallback, useEffect, useState } from "react";
+
+import { useStyles2 } from '@grafana/ui';
+import { getStyles } from "app/percona/add-instance/components/AddRemoteInstance/FormParts/FormParts.styles";
+import { GET_NODES_CANCEL_TOKEN } from "app/percona/inventory/Inventory.constants";
+import { AgentsOptions, NodesOptions } from "app/percona/inventory/Inventory.types";
+import { SelectField } from "app/percona/shared/components/Form/SelectFieldCore";
+import { useCancelToken } from "app/percona/shared/components/hooks/cancelToken.hook";
+import { fetchNodesOptionsAction } from "app/percona/shared/core/reducers/nodes/nodes";
+import { isApiCancelError } from "app/percona/shared/helpers/api";
+import { logger } from "app/percona/shared/helpers/logger";
+import { useAppDispatch } from "app/store/store";
+
+export interface NodesAgentsProps {
+  form?: FormApi;
+}
+
+export const NodesAgents: FC<NodesAgentsProps> = ({ form }) => {
```

Comment:
Also, please also write unit tests for the component so we can check that the logic around node/agent selection works as expected.